src/support/bits.cpp

```diff
@@ -188,7 +188,12 @@ uint32_t log2(uint32_t v) {
   return 31 - countLeadingZeroes(v);
 }
 
-uint32_t pow2(uint32_t v) { return 1 << v; }
+uint32_t pow2(uint32_t v) {
+  if (v >= sizeof(uint32_t) * CHAR_BIT) {
```

Comment:
This is to avoid undefined behavior, I assume?

Instead of halting the program, maybe it could handle overflow the way addition and multiplication do, that is, just leave the non-overflowed part? That is, if `v` is too big, it returns 0?

Reply:
Yes. Usually if we got `v >= 32` result becomes zero but may be undefined in some cases. I'm not sure it will be right wrap it as `v = v & 31` or something like this. Usually it mean something really unexpected happened

Reply:
Another possible solution is to replace `unreachable` to `assert`. wdyt?

Reply:
Personally I think it's ok to overflow here. It's like an overflow of an add, or a multiply? All those overflow silently.

Reply:
Ok, I just wrap `v` as `1 << (v & 31)` which remove UB